astropy/table/tests/test_column.py

```diff
@@ -464,6 +465,28 @@ def test_mask_on_non_masked_table(self):
             t["a"].mask = [True, False]
 
 
+@pytest.mark.parametrize(
+    "data",
+    [
+        np.array([1], dtype=np.int32),
```

Comment:
Maybe keep the test focussed just on `object`? I'd do `np.array([1], dtype=object)` and `[object()]` only.